src/neomaril_codex/training.py

```diff
@@ -14,6 +17,34 @@
 
 patt = re.compile(r'(\d+)')
 
+class NeomarilTrainingLogger:
+    def __init__(self, name, X_train, y_train):
+        self.name = name
+        self.X_train = X_train
+        self.y_train = y_train
+        self.model_outputs = None
+        self.model = None
+        self.metrics = {}
+        self.params = {}
+        self.requirements = None
+        self.python_version = None
+        self.extras = []
```

Comment:
We should have some methods for those, i think something to save plots, extra dataframes and something more generic.

Reply:
Implemented save_plot and a part of add_extra to work with dataframe.